### PR TITLE
feat: workforce identity federation

### PR DESCRIPTION
**Pull request by lsirac**
- Adds workforce support to pluggable auth w/ ADC
- Documents workforce in the README, as it is now in [preview](https://cloud.google.com/iam/docs/workforce-identity-federation)

### CODE REVIEW

**Comment by aitemour79** [comment]
Okay gracias por la invitación de cumpleaños para mi cumple al Cliente no
me acuerdo bien gracias

fun plus bouddi Mohamed

في الأربعاء، 3 غشت، 2022 02:15 Leo ***@***.***> كتب:

**Comment by aitemour79** [comment]
ok

fun plus bouddi Mohamed

في الأربعاء، 3 غشت، 2022 02:15 Leo ***@***.***> كتب:

**Comment by lsirac** [comment]
@siliconmoo

File: README.md [hunk 1]
```diff
@@ -458,13 +462,207 @@ credentials unless they do not meet your specific requirements.
...
-#### Using External Identities
+###  Workforce Identity Federation
 
-External identities (AWS, Azure, and OIDC-based providers) can be used with
-`Application Default Credentials`. In order to use external identities with Application Default
-Credentials, you need to generate the JSON credentials configuration file for your external identity
-as described above. Once generated, store the path to this file in the
-`GOOGLE_APPLICATION_CREDENTIALS` environment variable.
+[Workforce identity federation](https://cloud.google.com/iam/docs/workforce-identity-federation) lets you use an
+external identity provider (IdP) to authenticate and authorize a workforce—a group of users, such as employees,
```
Comment (TimurSadykov) [inline README.md:468]: spaces around "-"?

File: README.md [hunk 2]
```diff
@@ -458,13 +462,207 @@ credentials unless they do not meet your specific requirements.
...
+###  Workforce Identity Federation
 
-External identities (AWS, Azure, and OIDC-based providers) can be used with
-`Application Default Credentials`. In order to use external identities with Application Default
-Credentials, you need to generate the JSON credentials configuration file for your external identity
-as described above. Once generated, store the path to this file in the
-`GOOGLE_APPLICATION_CREDENTIALS` environment variable.
+[Workforce identity federation](https://cloud.google.com/iam/docs/workforce-identity-federation) lets you use an
+external identity provider (IdP) to authenticate and authorize a workforce—a group of users, such as employees, 
+partners, and contractors—using IAM, so that the users can access Google Cloud services. Workforce identity federation
```
Comment (TimurSadykov) [inline README.md:469]: also spaces around "-"? Or is it some formatting?

**Review by TimurSadykov — APPROVED** [review]
One maybe formatting fix, LGTM otherwise

**Pull request merged**